Replace connect-gzip-static with express-static-gzip to become compatible with node 23.

bdolgov opened the pull request:
Removes a dependency on `@folder/readdir` which is not compatible with node 23.

Fixes #24568.

Koenkk commented:
Connect-gzip-static > 3 requires Node 20, but unfortunately we cannot upgrade due to https://github.com/nodejs/docker-node/issues/1946 . I hope it will be fixed soon.

Koenkk commented:
Given that Node 18 is in maintenance until +- April 2025, we also need to support it until then (since that is the strategy z2m follows). Of course we also want to support Node 23. Therefore I think the best way forward is to remove this dependency, could you take a look into that?

![image](https://github.com/user-attachments/assets/3acb03c1-08eb-4e97-b8b9-9935c09222f7)

bdolgov committed:
fix: replace connect-gzip-static with express-static-gzip.

Old versions of connect-gzip-static are not compatible with node 23,
new versions of connect-gzip-static are not compatible with node 18,
but we want to support both. express-static-gzip is compatible with
all versions.

Koenkk commented:
Thanks!
